### DISCUSSION TITLE
api.delete withi filter

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#21201] [Feature Requests]
### Summary

In an extension, how use api.delete with filters when id is not known?
Do we have first to get all the ids, or can we include the filter directly in the api.delete?

### Basic Example

api.delete('/items/mouvements?filter={"email":"' + email.value + '"}') seems to have no effects

### Motivation

concision

### Detailed Design

api.delete('/items/mouvements?filter={"email":"' + email.value + '"}')

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

-

**Comment by br41nslug** [comment]
In a backend extension you'd use the `ItemsService.deleteByQuery()` https://github.com/br41nslug/directus-extension-reference/blob/main/reference/classes/class.ItemsService.md#deletebyquery
And for frontend extensions (or external apps, anything using the API really) you can delete many items using a query https://docs.directus.io/reference/items.html#delete-multiple-items
